WalletWasabi/Models/UpdateStatus.cs

```diff
@@ -21,9 +23,9 @@ public UpdateStatus(bool backendCompatible, bool clientUpToDate)
 
 		public bool Equals(UpdateStatus other) => this == other;
 
-		public override int GetHashCode() => ClientUpToDate.GetHashCode() ^ BackendCompatible.GetHashCode();
+		public override int GetHashCode() => (13 * ClientUpToDate.GetHashCode()) ^ (47 * BackendCompatible.GetHashCode()) ^ LegalDocumentsVersion.GetHashCode();
```

Comment:
I don't like magic numbers. Can we just replace it with this? 

```suggestion
		public override int GetHashCode() => string.Concat(ClientUpToDate, BackendCompatible, LegalDocumentsVersion).GetHashCode();
```

Reply:
The better idea you can do this with $ operator too.

Reply:
??? @lontivero

Reply:
This is possible, it will work and I am not against it, but I would prefer to implement it as microsoft does it in their own code. This calls calls the `bool.ToString()` for the two values and then concatenates it to get the hash code, it is not that it is slower and allocates memory but just that it is not "the best practice". A good recommendation is [this one](https://stackoverflow.com/a/263416/627071) by Jon Skeet.

Reply:
How about this? https://stackoverflow.com/a/4630550/1844347

Reply:
wow, I am getting old. That solution looks good. 
```c#
var v = new Version(1,1,0);
(true, false, v).GetHashCode()  // -1909889779
(false, false, v).GetHashCode() //-1909888690
(false, true, v).GetHashCode()  // -1909888723
(true, v).GetHashCode()         //-1360399228
(true, true, v).GetHashCode()   //-1909889746
(v).GetHashCode()               //  269488127
(v, true, true).GetHashCode()   // -230938503
(v, true, false).GetHashCode()  // -230938504
(v, false, false).GetHashCode() // -230938471
(v, false, true).GetHashCode()  // -230938472   (this is a bit annoying because clearly the last element is not multiplied by any factor)
```

Well, it looks good. The problem is that I don't know how it does it, it could be using the string representation as your previous suggestion (what is okay and works okay of course). I like this one but I hate the hidden magic even more than the magic numbers. This one also creates a new instance every time something invokes the GetHashCode. 

The good part is that this one provides us an uniform way to implement the GetHashCode method, what is great. I am not against this one as I am not against the implementation that uses strings.

Reply:
It doesn't provide a uniform way to implement `GetHashCode` as many of the times you don't have string representations.

Reply:
It is not using the ToString to create the hash. It uses `HashHelpers.Combine()`

https://github.com/dotnet/roslyn/blob/4ca74a1cde4165989c3c08272b6e56f511beaedc/src/Compilers/Test/Resources/Core/NetFX/ValueTuple/ValueTuple.cs#L297